src/Marten/Schema/DocumentMapping.cs

```diff
@@ -165,12 +165,24 @@ public PropertySearching PropertySearching
 
         public static string TableNameFor(Type documentType)
         {
-            return "mt_doc_" + documentType.Name.ToLower();
+            return "mt_doc_" + DocumentPersistentName(documentType);
         }
 
         public static string UpsertNameFor(Type documentType)
         {
-            return "mt_upsert_" + documentType.Name.ToLower();
+            return "mt_upsert_" + DocumentPersistentName(documentType);
+        }
+
+        private static string DocumentPersistentName(Type documentType)
+        {
+            var parts = new List<string>();
+            parts.Add(documentType.Name.ToLower());
+            if (documentType.IsNested)
```

Comment:
not sure if we'd want to go recursively to another level of nesting or now. for now I figured one level deep was sufficient.